pkg/storage/tpr/storage_versioner.go

```diff
@@ -103,32 +103,23 @@ func updateState(v storage.Versioner, st *objState, userUpdate storage.UpdateFun
 	return ret, ttl, nil
 }
 
-// UpdateList sets the resource version into an API list object. Returns an error if the object
-// cannot be updated correctly. May return nil if the requested object does not need metadata
-// from database.
-func (t *versioner) UpdateList(obj runtime.Object, resourceVersion uint64) error {
-	// ns, err := GetNamespace(obj)
-	// if err != nil {
-	// 	return err
-	// }
-	// if rvErr := GetAccessor().SetResourceVersion(
-	// 	obj,
-	// 	strconv.Itoa(int(resourceVersion)),
-	// ); rvErr != nil {
-	// 	return rvErr
-	// }
-	// unstruc, err := ToUnstructured(obj)
-	// if err != nil {
-	// 	return err
-	// }
-	// cl, err := GetResourceClient(t.cl, t.listKind, ns)
-	// if err != nil {
-	// 	return err
-	// }
-	// if _, err := cl.Update(unstruc); err != nil {
-	// 	return err
-	// }
-	return errors.New("UpdateList not implemented")
+// UpdateList receives a list object and ranges over constituent objects,
+// calling UpdateObject for each.
+func (v *versioner) UpdateList(listObj runtime.Object, listResourceVersion uint64) error {
+	err := meta.EachListItem(listObj, func(obj runtime.Object) error {
+		objResourceVersion, err := v.ObjectResourceVersion(obj)
+		if err != nil {
+			return err
+		}
+		if err = v.UpdateObject(obj, objResourceVersion); err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
```

Comment:
maybe just squash these 4 lines into:
return err
unless you're planning on doing something in the err condition loop?